internal/pkg/manifest/workload.go

```diff
@@ -149,7 +149,8 @@ func (b *BuildArgsOrString) UnmarshalYAML(unmarshal func(interface{}) error) err
 	}
 
 	if !b.BuildArgs.isEmpty() {
-		// Unmarshaled successfully to b.BuildArgs, return.
+		// Unmarshaled successfully to b.BuildArgs, unset b.BuildString, and return.
+		b.BuildString = nil
```

Comment:
does this UnmarshalYAML have unit tests? can we do something similar to above to make sure that the buildstring gets reset to nil if a custom build field is provided

Reply:
it is tested here https://github.com/aws/copilot-cli/pull/1686/files#diff-c4a6f6a13adfd37f7bd3fbc2d6d56fe5d1f2a6e752f6d795628adca336bf97e3R157. since `newDefaultLoadBalancedWebService` has default to be "/" and it is set back to nil

Reply:
wut am I misunderstanding something? that one is for UnmarshalYAML for healthchecks whereas this change is for the `image.build` field

Reply:
oooh sorry i misread it.